src/test/java/net/snowflake/ingest/streaming/internal/IcebergDataTypeParserTest.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2024 Snowflake Computing Inc. All rights reserved.
+ */
+
+package net.snowflake.ingest.streaming.internal;
+
+import java.util.ArrayList;
+import java.util.List;
+import net.snowflake.ingest.utils.IcebergDataTypeParser;
+import org.apache.iceberg.types.Type;
+import org.apache.iceberg.types.Types;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+/** Test for Iceberg data type serialization and deserialization. */
+public class IcebergDataTypeParserTest {
+  private class DataTypeInfo {
+
+    // Json representation of Iceberg data type
+    String jsonStr;
+    Type icebergType;
+
+    DataTypeInfo(String jsonStr, Type icebergType) {
+      this.jsonStr = jsonStr;
+      this.icebergType = icebergType;
+    }
+  }
+
+  private int fieldId = 0;
+
+  List<DataTypeInfo> dataTypesToTest;
+
+  @Before
+  public void setup() {
+    fieldId = 0;
+
+    // Create a Iceberg data type information list with primitive types and nested types.
+    dataTypesToTest = new ArrayList<>();
+    dataTypesToTest.add(new DataTypeInfo("\"boolean\"", Types.BooleanType.get()));
+    dataTypesToTest.add(new DataTypeInfo("\"int\"", Types.IntegerType.get()));
+    dataTypesToTest.add(new DataTypeInfo("\"long\"", Types.LongType.get()));
+    dataTypesToTest.add(new DataTypeInfo("\"float\"", Types.FloatType.get()));
+    dataTypesToTest.add(new DataTypeInfo("\"double\"", Types.DoubleType.get()));
+    dataTypesToTest.add(new DataTypeInfo("\"string\"", Types.StringType.get()));
+    dataTypesToTest.add(new DataTypeInfo("\"date\"", Types.DateType.get()));
+    dataTypesToTest.add(new DataTypeInfo("\"time\"", Types.TimeType.get()));
+    dataTypesToTest.add(new DataTypeInfo("\"timestamptz\"", Types.TimestampType.withZone()));
+    dataTypesToTest.add(
```

Comment:
the structured data type testing needs to cover more cases; and also account for nested schema parsing.

Reply:
Added to [jira](https://snowflakecomputing.atlassian.net/browse/SNOW-1490151).